File: quasar/Quasar.DataTransfer/Vendas/Request/VendaInserirRequest.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace Quasar.DataTransfer.Vendas.Request
{
    public class VendaInserirRequest
    {
        public int CodEndereco {get; set;}
        public int CodFormaPagamento {get; set;}
        public string? CodUsuario {get; set;}
        public int CodStatusVenda {get; set;}
        public IList<ItemVendaInserirRequest> Itens {get; set;}
    }
}

File: quasar-front/src/app/vendas/models/requests/vendaInserir.request.ts
import { ItemVendaInserirRequest } from './itemVendaInserir.request';

export class VendaInserirRequest {
  public codEndereco: number;
  public codFormaPagamento: number;
  public codStatusVenda: number;
  public itens: ItemVendaInserirRequest[];

  constructor(params: Partial<VendaInserirRequest>) {
    this.codEndereco = params.codEndereco || 0;
    this.codFormaPagamento = params.codFormaPagamento || 0;
    this.codStatusVenda = params.codStatusVenda || 0;
    this.itens = params.itens || [];
  }
}
